tools/utils/http-helpers.js

```diff
@@ -358,6 +369,47 @@ _.extend(exports, {
     } else {
       return body;
     }
-  }
+  },
 
+  getUrlWithResuming(urlOrOptions) {
+    const options = _.isObject(urlOrOptions) ? urlOrOptions : {
+      url: urlOrOptions,
+    };
+
+    const outputStream = new WritableStreamBuffer();
+
+    const MAX_ATTEMPTS = 5;
+    function attempt(triesRemaining) {
+      const headers = _.extend({}, urlOrOptions.headers, {
+        range: `bytes=${outputStream.size()}-`,
+      });
+      try {
+        return httpHelpers.request({
+          outputStream,
+          headers,
+          ...urlOrOptions,
+        });
+      } catch (e) {
+        console.log(e);
+        if (triesRemaining > 0) {
+          console.log(`Request failed, retrying`);
+          return attempt(triesRemaining - 1);
+        } else {
+          console.log(`Request failed ${MAX_ATTEMPTS} times, failing`)
+          throw new files.OfflineError(e);
+        }
+      }
+    }
+    const response = attempt(MAX_ATTEMPTS).response;
+    const body = outputStream.getContents();
+
+    if (response.statusCode >= 400 && response.statusCode < 600) {
+      const href = response.request.href;
+      throw Error(
+         body ||
+          `Could not get ${href}; server returned [${response.statusCode}]`);
```

Comment:
Maybe this was just for debugging, but I would use the same error message every time, regardless of whether `body` is defined, and then just `return outputStream.getContents()` below, if there was no error.